virttest/utils_iptables.py

```diff
@@ -84,3 +87,198 @@ def setup_or_cleanup_iptables_rules(cls, rules, params=None,
         # cleanup server session
         if params:
             server_session.close()
+
+
+class Firewalld(object):
+    """
+    class handle firewalld service
+    """
+    def __init__(self, session=None):
+        """
+        Initialize firewalld service by installing and creating service obj
+
+        :param session: ShellSession Object of guest/remote host
+        """
+        self.service_name = "firewalld"
+        if not utils_package.package_install(self.service_name, session=session):
+            logging.error("Firewalld package is not available")
+        if session:
+            self.session = session
+            self.firewalld = "service firewalld"
+        else:
+            self.firewalld = service.Factory.create_service("firewalld")
+
+        if not self.status():
+            self.start()
+
+    def status(self):
+        """
+        Method to get the status of firewalld service
+        """
+        if self.session:
+            cmd = "%s status" % self.firewalld
+            status, output = self.session.cmd_status_output(cmd)
+            if status == 0:
+                return bool(re.search(r"running", output))
+            else:
+                logging.error(output)
+                return False
+        else:
+            return self.firewalld.status()
+
+    def start(self):
+        """
+        Method to start firewalld service
+        """
+        if self.session:
+            cmd = "%s start" % self.firewalld
+            status, output = self.session.cmd_status_output(cmd)
+            if status != 0:
+                logging.error(output)
+        else:
+            self.firewalld.start()
+
+    def stop(self):
+        """
+        Method to stop firewalld service
+        """
+        if self.session:
+            cmd = "%s stop" % self.firewalld
+            status, output = self.session.cmd_status_output(cmd)
+            if status != 0:
+                logging.error(output)
+        else:
+            self.firewalld.stop()
+
+    def restart(self):
+        """
+        Method to restart firewalld service
+        """
+        if self.session:
+            cmd = "%s restart" % self.firewalld
+            status, output = self.session.cmd_status_output(cmd)
+            if status != 0:
+                logging.error(output)
+        else:
+            self.firewalld.restart()
+
+
+class Firewall_cmd(Firewalld):
```

Comment:
can be make Firewall_cmd singleton? in this way it avoid create multiple Firewall_cmd